modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/Header.scala

```diff
@@ -0,0 +1,35 @@
+package edu.gemini.seqexec.server
+
+import edu.gemini.seqexec.model.dhs.ObsId
+import edu.gemini.seqexec.server.DhsClient._
+
+/**
+  * Created by jluhrs on 1/31/17.
+  */
+trait Header {
+  def sendBefore(id: ObsId, inst: String): SeqAction[Unit]
+  def sendAfter(id: ObsId, inst: String): SeqAction[Unit]
+}
+
+object Header {
+  def buildKeyword[A](get: SeqAction[A], name: String, f: (String, A) => DhsClient.Keyword[A]): KeywordBag => SeqAction[KeywordBag] = k =>
+      get.map(x => k.add(f(name, x)))
+  def buildInt8(get: SeqAction[Byte], name: String ): KeywordBag => SeqAction[KeywordBag]       = buildKeyword(get, name, Int8Keyword)
+  def buildInt16(get: SeqAction[Short], name: String ): KeywordBag => SeqAction[KeywordBag]     = buildKeyword(get, name, Int16Keyword)
+  def buildInt32(get: SeqAction[Int], name: String ): KeywordBag => SeqAction[KeywordBag]       = buildKeyword(get, name, Int32Keyword)
+  def buildFloat(get: SeqAction[Float], name: String ): KeywordBag => SeqAction[KeywordBag]     = buildKeyword(get, name, FloatKeyword)
+  def buildDouble(get: SeqAction[Double], name: String ): KeywordBag => SeqAction[KeywordBag]   = buildKeyword(get, name, DoubleKeyword)
+  def buildBoolean(get: SeqAction[Boolean], name: String ): KeywordBag => SeqAction[KeywordBag] = buildKeyword(get, name, BooleanKeyword)
+  def buildString(get: SeqAction[String], name: String ): KeywordBag => SeqAction[KeywordBag]   = buildKeyword(get, name, StringKeyword)
+
+  private def bundleKeywords(inst: String, ks: Seq[KeywordBag => SeqAction[KeywordBag]]): SeqAction[KeywordBag] = {
```

Comment:
Because most of the time the action of retrieving a value involves reading from an EPICS channel, and it can fail.